### ISSUE TITLE
Incorrect definite assignment error with local functions

### ISSUE THREAD

**Issue by gafter** [dotnet/roslyn#13739]
The following program produces an incorrect error.

```cs
using System;

class Program
{
    public static void Main(string[] args)
    {
        int u;
        M();
    https://github.com/dotnet/roslyn/issues/13739
        Console.WriteLine(u); // error: use of unassigned local variable 'u'
        return;

        void M()
        {
            goto La;
        Lb: return;
        La: u = 3;
            goto Lb;
        }
    }
}
```

**Comment by agocke** [comment]
@gafter Ha! Nice trick using the link as a label.

I guess the problem is we don't trim unreferenced labels from the pending branches?

**Comment by gafter** [comment]
The problem is that we don't initialize the initial value of the "written by local function" set to `Unreachable()`.

**Comment by agocke** [comment]
@gafter Agreed.

On the plus side, this is an error in precision, not in soundness, so no unsafe program can be compiled because of this bug.

**Comment by agocke** [comment]
@dotnet/roslyn-compiler Could one more person take a look at this please?

**Comment by agocke** [comment]
Woops, wrong issue :)

**Comment by agocke** [comment]
Resolved by #14190
